README.md

```diff
@@ -109,6 +109,49 @@ The text to be included in the PR comment.
 
 **Required**: true
 
+##### Templates
+
+Comment snippet bodies (as well as `comment.header` and `comment.footer`) are [Mustache][https://mustache.github.io/mustache.5.html] templates.
+
+Variables for the template can be provided via the `template-variables` input which should be a string containing a valid JSON.
+
+You can use the [context and expression syntax](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions) to assemble the JSON and [set-output](https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-output-parameter) to calculate data for the template in separate steps.
+
+###### Example
+
+```yaml
+name: "PR Commenter"
+on:
+  - pull_request_target
+
+jobs:
+  pr-comment:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Calculate some template variables
+        id: vars
+        run: |
+          echo ::set-output name=today::"$(date +%d-%m-%Y)"
+
+      - uses: exercism/pr-commenter-action@v1.1.0
+        with:
+          template-variables: |
+            {
+              "today": "${{ steps.vars.outputs.today }}",
+              "prAuthor": "${{ github.event.pull_request.user.login }}",
+              "branchNamePrefix": ${{ startsWith(github.event.pull_request.head.ref, 'ref-') }}
+            }
```

Comment:
Is there a nicer way to pass an object as input?

Reply:
I don't know of one.

You could pass them as environment variables with a predetermined prefix, e.g.

```yaml
env:
  VAR_PR_AUTHOR: ${{ github.event.pull_request.user.login }}
  VAR_TODAY: ${{ steps.vars.outputs.today }}
  VAR_BRANCH_NAME_PREFIX: ${{ startsWith(github.event.pull_request.head.ref, 'ref-') }}
```

or, if you don't want a prefix, allow all environment variables as inputs for the template.